# Proyecto No. 1 Data Science
## Universidad del Valle de Guatemala Segundo Semestre 2021

#### Andrés Say, 19705 Ayleen Rubio, 19003 Diego Alegría, 15171 Mario Sarmientos, 17055

# Importación de liberías

In [1]:
import pandas as pd
import numpy as np
import os 

## Lectura de directorio y unificación de Data Frame

In [2]:
files=os.listdir('.\EstablecimientosMINEDUC')

In [3]:
df=pd.read_csv('.\EstablecimientosMINEDUC\\'+files[0],encoding = "ISO-8859-1",header=1,skipfooter=2,engine='python')
df=pd.DataFrame(columns=df.columns)
for i in files:
    temp=(pd.read_csv('.\EstablecimientosMINEDUC\\'+i,encoding = "ISO-8859-1",header=1,skipfooter=2,engine='python'))
    
    temp.loc[temp.DISTRITO=='','DISTRITO']=np.nan
    temp.loc[temp.DEPARTAMENTO=='','DEPARTAMENTO']=np.nan
    temp.loc[temp.MUNICIPIO=='','MUNICIPIO']=np.nan
    temp.loc[temp.DIRECCION=='nan','DIRECCION']=np.nan
    temp.loc[temp.TELEFONO=='nan','TELEFONO']=np.nan
    temp.loc[temp.ESTABLECIMIENTO=='nan','ESTABLECIMIENTO']=np.nan
    temp.loc[temp.DIRECCION=='','DIRECCION']=np.nan
    temp.loc[temp.TELEFONO=='','TELEFONO']=np.nan
    
    
    temp.DEPARTAMENTO=temp.DEPARTAMENTO.fillna(method='ffill')
    
    df=df.append(temp,ignore_index=True)
    
df
    

CODIGO DISTRITO  DEPARTAMENTO  MUNICIPIO  \
0     16-01-0138-46   16-031  ALTA VERAPAZ      COBAN   
1     16-01-0139-46   16-031  ALTA VERAPAZ      COBAN   
2     16-01-0140-46   16-031  ALTA VERAPAZ      COBAN   
3     16-01-0141-46   16-005  ALTA VERAPAZ      COBAN   
4     16-01-0142-46   16-005  ALTA VERAPAZ      COBAN   
...             ...      ...           ...        ...   
7959  19-09-0040-46   19-026        ZACAPA   LA UNION   
7960  19-09-0902-46   19-026        ZACAPA   LA UNION   
7961  19-10-0013-46   19-016        ZACAPA      HUITE   
7962  19-10-1009-46   19-016        ZACAPA      HUITE   
7963  19-11-0018-46   19-020        ZACAPA  SAN JORGE   

                                        ESTABLECIMIENTO  \
0                                         COLEGIO COBAN   
1                      COLEGIO PARTICULAR MIXTO VERAPAZ   
2                               COLEGIO "LA INMACULADA"   
3              ESCUELA NACIONAL DE CIENCIAS COMERCIALES   
4     INSTITUTO NORMAL MIXTO DEL NORTE 'EMILIO ROSAL...   
...                                                 ...   
7959                       LICEO PARTICULAR MIXTO JIREH   
7960         LICEO PARTICULAR MIXTO JIREH (POR MADUREZ)   
7961                            INSTITUTO DIVERSIFICADO   
7962  INSTITUTO MIXTO DE EDUCACION DIVERSIFICADA POR...   
7963  INSTITUTO MIXTO DE EDUCACIÓN DIVERSIFICADA POR...   

                                   DIRECCION    TELEFONO  \
0     KM.2 SALIDA A SAN JUAN CHAMELCO ZONA 8    77945104   
1               KM 209.5 ENTRADA A LA CIUDAD    77367402   
2                  7A. AVENIDA 11-109 ZONA 6    49532425   
3                      2A CALLE 11-10 ZONA 2    79514215   
4                        3A AVE 6-23 ZONA 11    79521468   
...                                      ...         ...   
7959                            BARRIO NUEVO  79418369.0   
7960                            BARRIO NUEVO  79418369.0   
7961                     BARRIO BUENOS AIRES  31863599.0   
7962                         BARRIO EL CAMPO  55958103.0   
7963                        BARRIO EL CENTRO  41447589.0   

                               SUPERVISOR  \
0           CESAR ROLANDO CHOCOOJ YALIBAT   
1           CESAR ROLANDO CHOCOOJ YALIBAT   
2           CESAR ROLANDO CHOCOOJ YALIBAT   
3                 JUAN CARLOS WINTER CAAL   
4                 JUAN CARLOS WINTER CAAL   
...                                   ...   
7959  BERTA ALICIA LEIVA CORDON DE GARCIA   
7960  BERTA ALICIA LEIVA CORDON DE GARCIA   
7961         WENDER HUMBERTO MATEO NAJERA   
7962         WENDER HUMBERTO MATEO NAJERA   
7963                      ALBA LUZ MENDEZ   

                                 DIRECTOR          NIVEL       SECTOR    AREA  \
0               JULIO CESAR VILLELA AMADO  DIVERSIFICADO      PRIVADO  URBANA   
1                                     NaN  DIVERSIFICADO      PRIVADO  URBANA   
2                  MERCEDES QUIROS QUIROS  DIVERSIFICADO      PRIVADO  URBANA   
3                                     NaN  DIVERSIFICADO      OFICIAL  URBANA   
4                                     NaN  DIVERSIFICADO      OFICIAL  URBANA   
...                                   ...            ...          ...     ...   
7959  ANA MARÍA CUELLAR GUERRA DE RAMÍREZ  DIVERSIFICADO      PRIVADO  URBANA   
7960  ANA MARÍA CUELLAR GUERRA DE RAMÍREZ  DIVERSIFICADO      PRIVADO  URBANA   
7961            ERIC MANUEL PÉREZ AGUILAR  DIVERSIFICADO      OFICIAL  URBANA   
7962            ROBIDIO PORTILLO SALGUERO  DIVERSIFICADO  COOPERATIVA  URBANA   
7963            VICTOR HUGO GUERRA MONROY  DIVERSIFICADO  COOPERATIVA  URBANA   

       STATUS   MODALIDAD     JORNADA             PLAN DEPARTAMENTAL  
0     ABIERTA  MONOLINGUE    MATUTINA  DIARIO(REGULAR)  ALTA VERAPAZ  
1     ABIERTA  MONOLINGUE    MATUTINA  DIARIO(REGULAR)  ALTA VERAPAZ  
2     ABIERTA  MONOLINGUE    MATUTINA  DIARIO(REGULAR)  ALTA VERAPAZ  
3     ABIERTA  MONOLINGUE    MATUTINA  DIARIO(REGULAR)  ALTA VERAPAZ  
4     ABIERTA    BILINGUE  VE

## Lista de variables (Columnas)

In [4]:
display(df.columns.tolist())

['CODIGO',
 'DISTRITO',
 'DEPARTAMENTO',
 'MUNICIPIO',
 'ESTABLECIMIENTO',
 'DIRECCION',
 'TELEFONO',
 'SUPERVISOR',
 'DIRECTOR',
 'NIVEL',
 'SECTOR',
 'AREA',
 'STATUS',
 'MODALIDAD',
 'JORNADA',
 'PLAN',
 'DEPARTAMENTAL']

## Tipos de variables

In [5]:
df.dtypes

CODIGO             object
DISTRITO           object
DEPARTAMENTO       object
MUNICIPIO          object
ESTABLECIMIENTO    object
DIRECCION          object
TELEFONO           object
SUPERVISOR         object
DIRECTOR           object
NIVEL              object
SECTOR             object
AREA               object
STATUS             object
MODALIDAD          object
JORNADA            object
PLAN               object
DEPARTAMENTAL      object
dtype: object

## Análisis para variable Código

#### Revisión de Formato

In [6]:
display(df.CODIGO.unique())

array(['16-01-0138-46', '16-01-0139-46', '16-01-0140-46', ...,
       '19-10-0013-46', '19-10-1009-46', '19-11-0018-46'], dtype=object)

#### Revisión de Nulls

In [7]:
# Utilizando el criterio de pandas
display(df.loc[df.CODIGO.isna(),:])

# Utilizando el criterio de la palabra nan

display(df.loc[df.CODIGO=='nan',:])


Empty DataFrame
Columns: [CODIGO, DISTRITO, DEPARTAMENTO, MUNICIPIO, ESTABLECIMIENTO, DIRECCION, TELEFONO, SUPERVISOR, DIRECTOR, NIVEL, SECTOR, AREA, STATUS, MODALIDAD, JORNADA, PLAN, DEPARTAMENTAL]
Index: []

Empty DataFrame
Columns: [CODIGO, DISTRITO, DEPARTAMENTO, MUNICIPIO, ESTABLECIMIENTO, DIRECCION, TELEFONO, SUPERVISOR, DIRECTOR, NIVEL, SECTOR, AREA, STATUS, MODALIDAD, JORNADA, PLAN, DEPARTAMENTAL]
Index: []

#### Revisión de duplicidad

In [8]:
a=len(df.CODIGO.value_counts())
b=len(df)

display(b-a)

#No existe duplicidad

0



Debido a que es un código que esta compuesto de la siguiente forma:

$$##-##-####-##$$ 

Donde # es un númeor entero del 0 al 9, debemos de revisar si todos tienen los números completos o existen casos donde no se cumpla el estándar antes mencionado. 

No se despliegan todos los valores puesto que cada código es único y no representa ningún tipo de relación con otros establecimientos más si presenta una relación con las otras variables de ubicación como departamento y municipio.



In [9]:
code=df.CODIGO.str.split('-',expand=True).copy()
code

0   1     2   3
0     16  01  0138  46
1     16  01  0139  46
2     16  01  0140  46
3     16  01  0141  46
4     16  01  0142  46
...   ..  ..   ...  ..
7959  19  09  0040  46
7960  19  09  0902  46
7961  19  10  0013  46
7962  19  10  1009  46
7963  19  11  0018  46

[7964 rows x 4 columns]

#### Revisión por partes del código de identificación

In [10]:
display(code.loc[code[0].isna(),:])
display(code.loc[code[1].isna(),:])
display(code.loc[code[2].isna(),:])
display(code.loc[code[3].isna(),:])

Empty DataFrame
Columns: [0, 1, 2, 3]
Index: []

Empty DataFrame
Columns: [0, 1, 2, 3]
Index: []

Empty DataFrame
Columns: [0, 1, 2, 3]
Index: []

Empty DataFrame
Columns: [0, 1, 2, 3]
Index: []

Podemos observar que todo se ve de forma correcta

## Análisis para variable Distrito

#### Revisión de Formato

In [11]:
display(df.DISTRITO.unique().tolist()[0:20])

['16-031',
 '16-005',
 '16-006',
 '16-037',
 '16-050',
 nan,
 '25-000',
 '16-053',
 '16-008',
 '16-007',
 '16-009',
 '16-011',
 '16-004',
 '16-010',
 '16-012',
 '16-036',
 '16-038',
 '16-013',
 '16-041',
 '16-042']

#### Revisión de Nulls

In [12]:
display(df.loc[df.DISTRITO.isna(),:])

CODIGO DISTRITO   DEPARTAMENTO                 MUNICIPIO  \
68    16-01-2947-46      NaN   ALTA VERAPAZ                     COBAN   
209   16-10-3816-46      NaN   ALTA VERAPAZ         SAN JUAN CHAMELCO   
344   15-03-0134-46      NaN   BAJA VERAPAZ                   RABINAL   
637   04-11-0995-46      NaN  CHIMALTENANGO                ACATENANGO   
674   04-15-0599-46      NaN  CHIMALTENANGO                  ZARAGOZA   
...             ...      ...            ...                       ...   
7641  10-03-2495-46      NaN  SUCHITEPEQUEZ  SAN FRANCISCO ZAPOTITLAN   
7712  10-13-0882-46      NaN  SUCHITEPEQUEZ                  CHICACAO   
7859  08-05-0480-46      NaN    TOTONICAPAN              MOMOSTENANGO   
7874  19-01-0083-46      NaN         ZACAPA                    ZACAPA   
7886  19-01-0166-46      NaN         ZACAPA                    ZACAPA   

                                        ESTABLECIMIENTO  \
68    INSTITUTO PRIVADO BASICO Y BACHILLERATO POR MA...   
209   LICEO DE BACHILLERATO EN COMPUTACION Y EDUCACI...   
344   LICEO PRIVADO MIXTO DE CIENCIAS COMERCIALES 'R...   
637               INSTITUTO PRACTICO DE EDUCACION MEDIA   
674   INSTITUTO PRIVADO MIXTO DE EDUCACION DIVERSIFI...   
...                                                 ...   
7641            INSTITUTO PRE-UNIVERSITARIO FRANCISCANO   
7712  INSTITUTO PRIVADO MIXTO DE EDUCACION DIVERSIFI...   
7859  INSTITUTO DIVERSIFICADO ADSCRITO AL INSTITUTO ...   
7874  INSTITUTO PRIVADO MIXTO DE EDUCACION DIVERSIFI...   
7886  INSTITUTO DE EDUCACION BASICA Y BACHILLERATO P...   

                    DIRECCION    TELEFONO SUPERVISOR  \
68      5A. CALLE 1-98 ZONA 3         NaN        NaN   
209                       NaN         NaN        NaN   
344   3A. AVENIDA 1-55 ZONA 3         NaN        NaN   
637            CANTON TERCERO         NaN        NaN   
674    6A. AVENIDA NORTE NO.2         NaN        NaN   
...                       ...         ...        ...   
7641    5A. CALLE 2-66 ZONA 1   5565-5011        NaN   
7712     5A. AVE. 1-76 ZONA 2         NaN        NaN   
7859          CASA PARROQUIAL         NaN        NaN   
7874        BARRIO LAS FLORES         NaN        NaN   
7886        ALDEA LOS JOCOTES  24106060.0        NaN   

                         DIRECTOR          NIVEL       SECTOR    AREA  \
68                            NaN  DIVERSIFICADO      PRIVADO  URBANA   
209                           NaN  DIVERSIFICADO      PRIVADO  URBANA   
344                           NaN  DIVERSIFICADO      PRIVADO  URBANA   
637                           NaN  DIVERSIFICADO  COOPERATIVA   RURAL   
674                           NaN  DIVERSIFICADO      PRIVADO  URBANA   
...                           ...            ...          ...     ...   
7641                          NaN  DIVERSIFICADO      PRIVADO  URBANA   
7712                          NaN  DIVERSIFICADO      PRIVADO  URBANA   
7859                          NaN  DIVERSIFICADO      PRIVADO  URBANA   
7874                          NaN  DIVERSIFICADO      PRIVADO  URBANA   
7886  FLOR DE MARIA LOPEZ BARRIOS  DIVERSIFICADO      OFICIAL   RURAL   

                     STATUS   MODALIDAD     JORNADA             PLAN  \
68    CERRADA TEMPORALMENTE  MONOLINGUE  VESPERTINA  DIARIO(REGULAR)   
209   CERRADA TEMPORALMENTE  MONOLINGUE    MATUTINA  DIARIO(REGULAR)   
344   CERRADA TEMPORALMENTE  MONOLINGUE  VESPERTINA  DIARIO(REGULAR)   
637   CERRADA TEMPORALMENTE  MONOLINGUE    NOCTURNA  DIARIO(REGULAR)   
674   CERRADA TEMPORALMENTE  MONOLINGUE    NOCTURNA  DIARIO(REGULAR)   
...                     ...         ...         ...              ...   
7641  CERRADA TEMPORALMENTE  MONOLINGUE  VESPERTINA  DIARIO(REGULAR)   
7712  CERRADA TEMPORALMENTE  MONOLINGUE       DOBLE  DIARIO(REGULAR)   
7859  CERRADA TEMPORALMENTE  MONOLINGUE  INTERMEDIA  DIARIO(REGULAR)   
7874  CERRADA TEMPORALMENTE  MONOLINGUE  VESPERTINA  DIARIO(REGULAR)   
7886  CERRADA TEMPORALMENTE  MONOLINGUE  VESPERTINA  DIARIO(REGULAR)   

      DEPART

#### Revisión de códigos únicos

In [13]:
df.DISTRITO.value_counts()

01-403    193
05-033    137
11-017    130
01-411    128
01-409     89
         ... 
18-027      1
18-009      1
12-034      1
18-007      1
19-020      1
Name: DISTRITO, Length: 674, dtype: int64




Debido a que es un Distrito que esta compuesto de la siguiente forma:

$$##-###$$ 

Donde # es un númeor entero del 0 al 9, debemos de revisar si todos tienen los números completos o existen casos donde no se cumpla el estándar antes mencionado. 

No se despliegan todos los valores puesto que cada código es único para el distrito


In [14]:
distrito=df.DISTRITO.str.split('-',expand=True)

#### Revisión por partes del código de identificación

In [15]:
print('Para primera parte del código de identificación del distrito')
display(distrito.loc[(distrito[0].isna())|(distrito[0]==''),:])
print('Para segunda parte del código de identificación del distrito')
display(distrito.loc[(distrito[1].isna())|(distrito[1]==''),:])

Para primera parte del código de identificación del distrito


0    1
68    NaN  NaN
209   NaN  NaN
344   NaN  NaN
637   NaN  NaN
674   NaN  NaN
...   ...  ...
7641  NaN  NaN
7712  NaN  NaN
7859  NaN  NaN
7874  NaN  NaN
7886  NaN  NaN

[212 rows x 2 columns]

Para segunda parte del código de identificación del distrito


0    1
68    NaN  NaN
209   NaN  NaN
344   NaN  NaN
637   NaN  NaN
674   NaN  NaN
...   ...  ...
7641  NaN  NaN
7712  NaN  NaN
7859  NaN  NaN
7874  NaN  NaN
7886  NaN  NaN

[245 rows x 2 columns]

Podemos notar que tenemos 212 establecimientos que no tienen un código de distrito en la primera parte del código y en la segunda tenemos 245 (los 212 anteriores + 33 de establecimientos que solo poseen la primera mitad del código de distrito)

#### Porcentaje de datos faltantes a corregir

In [16]:
a=len(distrito.loc[(distrito[1].isna())|(distrito[1]==''),:])
b=len(df)

print(str(round((a/b)*100,2))+' %')

3.08 %


#### Estrategia para corrección

Se plantea una función en donde se busca la coincidencia más cercana con tres criterios:
    
    - Departamento
    - Municipio
    - Región Departamental
    
Y se sustituye el espacio vacío utilizando programación funcional

In [17]:
def corregir_distrito(df,departamento,municipio,departamental):
    
    temp=df.loc[(df.DEPARTAMENTO==departamento)&(df.MUNICIPIO==municipio)&(df.DEPARTAMENTAL==departamental),'DISTRITO']
    return temp.value_counts()[0]

try:
    df.loc[(df.DISTRITO.isna())|(df.DISTRITO.str.split('-',expand=True)[1]==''),'DISTRITO'] = df.loc[(df.DISTRITO.isna())|(df.DISTRITO.str.split('-',expand=True)[1]==''),:].apply(lambda row : corregir_distrito(df,row['DEPARTAMENTO'],row['MUNICIPIO'],row['DEPARTAMENTAL']), axis = 1)
except:
    print('Todos los códigos ya han sido restaurados')

## Análisis para variable Departamento

#### Revisión de Formato

In [18]:
display(df.DEPARTAMENTO.unique())

array(['ALTA VERAPAZ', 'BAJA VERAPAZ', 'CHIMALTENANGO', 'CHIQUIMULA',
       'CIUDAD CAPITAL', 'EL PROGRESO', 'ESCUINTLA', 'GUATEMALA',
       'HUEHUETENANGO', 'IZABAL', 'JALAPA', 'JUTIAPA', 'PETEN',
       'QUETZALTENANGO', 'QUICHE', 'RETALHULEU', 'SACATEPEQUEZ',
       'SAN MARCOS', 'SANTA ROSA', 'SOLOLA', 'SUCHITEPEQUEZ',
       'TOTONICAPAN', 'ZACAPA'], dtype=object)

#### Revisión de Nulls

In [19]:
display(df.loc[df.DEPARTAMENTO.isna(),:])

Empty DataFrame
Columns: [CODIGO, DISTRITO, DEPARTAMENTO, MUNICIPIO, ESTABLECIMIENTO, DIRECCION, TELEFONO, SUPERVISOR, DIRECTOR, NIVEL, SECTOR, AREA, STATUS, MODALIDAD, JORNADA, PLAN, DEPARTAMENTAL]
Index: []

#### Formato a Proper Case 

In [20]:
df.DEPARTAMENTO=df.DEPARTAMENTO.str.title()

#### Conteo de establecimientos por departamento



In [21]:
df.DEPARTAMENTO.value_counts()

Ciudad Capital    1349
Guatemala         1237
Escuintla          496
San Marcos         486
Huehuetenango      437
Quetzaltenango     430
Peten              357
Suchitepequez      330
Izabal             303
Chimaltenango      290
Alta Verapaz       280
Sacatepequez       276
Jutiapa            265
Retalhuleu         255
Quiche             212
Chiquimula         168
Santa Rosa         140
Solola             138
Jalapa             116
Baja Verapaz       114
El Progreso        111
Zacapa              93
Totonicapan         81
Name: DEPARTAMENTO, dtype: int64

## Análisis para variable Municipio

#### Revisión de Formato

In [22]:
# Formato de datos
display(df.MUNICIPIO.unique().tolist()[0:10])

municipios=len(df.loc[~((df.DEPARTAMENTO=='Ciudad Capital')|(df.MUNICIPIO.str.contains('ZONA'))),'MUNICIPIO'].unique())
zonas=len(df.loc[(df.DEPARTAMENTO=='Ciudad Capital')|(df.MUNICIPIO.str.contains('ZONA')),'MUNICIPIO'].unique())

print('Cantidad de municipios con establecimientos en la base de datos: '+str(municipios))
print('Cantidad de zonas de la ciudad capital con establecimientos en la base de datos: '+str(zonas))

['COBAN',
 'SANTA CRUZ VERAPAZ',
 'SAN CRISTOBAL VERAPAZ',
 'TACTIC',
 'TAMAHU',
 'SAN MIGUEL TUCURU',
 'PANZOS',
 'SENAHU',
 'SAN PEDRO CARCHA',
 'SAN JUAN CHAMELCO']

Cantidad de municipios con establecimientos en la base de datos: 326
Cantidad de zonas de la ciudad capital con establecimientos en la base de datos: 21


#### Revisión de Nulls

In [23]:
display(df.loc[df.MUNICIPIO.isna(),:])

Empty DataFrame
Columns: [CODIGO, DISTRITO, DEPARTAMENTO, MUNICIPIO, ESTABLECIMIENTO, DIRECCION, TELEFONO, SUPERVISOR, DIRECTOR, NIVEL, SECTOR, AREA, STATUS, MODALIDAD, JORNADA, PLAN, DEPARTAMENTAL]
Index: []

#### Formato a Proper Case 

In [24]:
df.MUNICIPIO=df.MUNICIPIO.str.title()

#### Eliminación de espacios en blanco no necesarios

In [25]:
df.MUNICIPIO=df.MUNICIPIO.str.strip()

#### Eliminación de acentos para estandarización

In [26]:
df.MUNICIPIO=df.MUNICIPIO.str.replace('á','a')
df.MUNICIPIO=df.MUNICIPIO.str.replace('é','e')
df.MUNICIPIO=df.MUNICIPIO.str.replace('í','i')
df.MUNICIPIO=df.MUNICIPIO.str.replace('ó','o')
df.MUNICIPIO=df.MUNICIPIO.str.replace('ú','u')

#### Conteo de establecimientos por municipio o zona de la Ciudad Capital



In [27]:
df.MUNICIPIO.value_counts()

Zona 1                    554
Mixco                     356
Villa Nueva               302
Quetzaltenango            221
Zona 7                    154
                         ... 
Petatan                     1
Chiche                      1
San Francisco La Union      1
Zunil                       1
San Jorge                   1
Name: MUNICIPIO, Length: 347, dtype: int64

## Análisis para variable Establecimiento

#### Revisión de Formato

In [28]:
# Formato de datos
display(df.ESTABLECIMIENTO.unique().tolist()[0:10])


['COLEGIO COBAN',
 'COLEGIO PARTICULAR MIXTO VERAPAZ',
 'COLEGIO "LA INMACULADA"',
 'ESCUELA NACIONAL DE CIENCIAS COMERCIALES',
 "INSTITUTO NORMAL MIXTO DEL NORTE 'EMILIO ROSALES PONCE'",
 'COLEGIO PARTICULAR MIXTO IMPERIAL',
 'INSTITUTO DE TURSMO Y AVIACON DEL NORTE I.T.A.N',
 'INSTITUTO INTERCULTRUAL ALTAVERAPACENCESE -IIAV-',
 'LICEO "MODERNO LATINO"',
 'INSTITUTO NACIONAL DE EDUCACION DIVERSIFICADA']

#### Revisión de Nulls

In [29]:
display(df.loc[df.ESTABLECIMIENTO.isna(),:])

Empty DataFrame
Columns: [CODIGO, DISTRITO, DEPARTAMENTO, MUNICIPIO, ESTABLECIMIENTO, DIRECCION, TELEFONO, SUPERVISOR, DIRECTOR, NIVEL, SECTOR, AREA, STATUS, MODALIDAD, JORNADA, PLAN, DEPARTAMENTAL]
Index: []

#### Formato a Proper Case 

In [30]:
df.ESTABLECIMIENTO=df.ESTABLECIMIENTO.str.title()

#### Eliminación de espacios en blanco no necesarios

In [31]:
df.ESTABLECIMIENTO=df.ESTABLECIMIENTO.str.strip()

#### Eliminación de acentos para estandarización

In [32]:
df.ESTABLECIMIENTO=df.ESTABLECIMIENTO.str.replace('á','a')
df.ESTABLECIMIENTO=df.ESTABLECIMIENTO.str.replace('é','e')
df.ESTABLECIMIENTO=df.ESTABLECIMIENTO.str.replace('í','i')
df.ESTABLECIMIENTO=df.ESTABLECIMIENTO.str.replace('ó','o')
df.ESTABLECIMIENTO=df.ESTABLECIMIENTO.str.replace('ú','u')

#### Conteo de establecimientos por municipio o zona de la Ciudad Capital



In [33]:
df.ESTABLECIMIENTO.value_counts()

Instituto Nacional De Educacion Diversificada                                   398
Instituto De Educacion Diversificada Por Cooperativa De Enseñanza                30
Instituto Diversificado Por Cooperativa                                          24
Centro De Educacion Extraescolar -Ceex-                                          24
Instituto De Computacion Informatica                                             22
                                                                               ... 
Liceo La Juventud                                                                 1
Colegio Mixto Privado San Cris                                                    1
Liceo Inmaculado Corazon                                                          1
Instituto Privado Mixto De Educacion Diversificada Colegio Evangelico Hebron      1
Instituto Diversificado                                                           1
Name: ESTABLECIMIENTO, Length: 4551, dtype: int64

## Análisis para variable Dirección

#### Revisión de Formato

In [34]:
# Formato de datos
display(df.DIRECCION.unique().tolist()[0:10])


['KM.2 SALIDA A SAN JUAN CHAMELCO ZONA 8',
 'KM 209.5 ENTRADA A LA CIUDAD',
 '7A. AVENIDA 11-109 ZONA 6',
 '2A CALLE 11-10 ZONA 2',
 '3A AVE 6-23 ZONA 11',
 '5A. CALLE 1-9 ZONA 3',
 '3 AV. 5-28 ZONA 4',
 '7A. CALLE 11-09 ZONA 6 COBAN',
 '3A. AVAENIDA 1-23 ZONA 4',
 '11 AVENIDA 5-17 ZONA 4']

#### Revisión de Nulls

In [35]:
display(df.loc[df.DIRECCION.isna(),:].head())
a=len(df.loc[df.DIRECCION.isna(),:])
b=len(df)
print('Cantidad de establecimientos sin dirección: '+str(a)+' siendo el '+str(round((a/b)*100,2))+' % de toda la base de datos')

CODIGO DISTRITO    DEPARTAMENTO             MUNICIPIO  \
77   16-01-8813-46   25-000    Alta Verapaz                 Coban   
209  16-10-3816-46       14    Alta Verapaz     San Juan Chamelco   
695  20-01-0132-46   20-038      Chiquimula            Chiquimula   
809  20-08-1595-46   20-040      Chiquimula  Concepcion Las Minas   
940  00-01-0312-46   01-101  Ciudad Capital                Zona 1   

                                       ESTABLECIMIENTO DIRECCION TELEFONO  \
77   Instituto Guatemalteco De Educacion Radiofonic...       NaN      NaN   
209  Liceo De Bachillerato En Computacion Y Educaci...       NaN      NaN   
695     Escuela Privada De Educacion Fisica De Oriente       NaN      NaN   
809                           Liceo Tecnico Concepcion       NaN      NaN   
940                         Colegio Nacional Americano       NaN      NaN   

                                SUPERVISOR DIRECTOR          NIVEL   SECTOR  \
77           SONIA HAYDEE REYES MONTERROSO      NaN  DIVERSIFICADO  PRIVADO   
209                                    NaN      NaN  DIVERSIFICADO  PRIVADO   
695       ANGELA DEL ROSARIO GARCIA MARCOS      ---  DIVERSIFICADO  PRIVADO   
809                  LANDELINO CRUZ GARCIA      NaN  DIVERSIFICADO  PRIVADO   
940  MARIA ESTELA MARROQUIN CHUR DE ORTEGA      ---  DIVERSIFICADO  PRIVADO   

       AREA                 STATUS   MODALIDAD     JORNADA             PLAN  \
77    RURAL  CERRADA TEMPORALMENTE  MONOLINGUE    MATUTINA      A DISTANCIA   
209  URBANA  CERRADA TEMPORALMENTE  MONOLINGUE    MATUTINA  DIARIO(REGULAR)   
695  URBANA  CERRADA TEMPORALMENTE  MONOLINGUE  VESPERTINA  DIARIO(REGULAR)   
809  URBANA  CERRADA TEMPORALMENTE  MONOLINGUE  VESPERTINA  DIARIO(REGULAR)   
940  URBANA       TEMPORAL TITULOS  MONOLINGUE    MATUTINA  DIARIO(REGULAR)   

       DEPARTAMENTAL  
77      ALTA VERAPAZ  
209     ALTA VERAPAZ  
695       CHIQUIMULA  
809       CHIQUIMULA  
940  GUATEMALA NORTE

Cantidad de establecimientos sin dirección: 55 siendo el 0.69 % de toda la base de datos


#### Formato a Proper Case 

In [36]:
df.DIRECCION=df.DIRECCION.str.title()

#### Eliminación de espacios en blanco no necesarios

In [37]:
df.DIRECCION=df.DIRECCION.str.strip()

#### Eliminación de acentos para estandarización

In [38]:
df.DIRECCION=df.DIRECCION.str.replace('á','a')
df.DIRECCION=df.DIRECCION.str.replace('é','e')
df.DIRECCION=df.DIRECCION.str.replace('í','i')
df.DIRECCION=df.DIRECCION.str.replace('ó','o')
df.DIRECCION=df.DIRECCION.str.replace('ú','u')

#### Se agrega 'No hay dirección registrada' a los datos faltantes

In [39]:
df.loc[df.DIRECCION.isna(),'DIRECCION']='No hay dirección registrada'

display(df.loc[df.DIRECCION=='No hay dirección registrada',:].head())

CODIGO DISTRITO    DEPARTAMENTO             MUNICIPIO  \
77   16-01-8813-46   25-000    Alta Verapaz                 Coban   
209  16-10-3816-46       14    Alta Verapaz     San Juan Chamelco   
695  20-01-0132-46   20-038      Chiquimula            Chiquimula   
809  20-08-1595-46   20-040      Chiquimula  Concepcion Las Minas   
940  00-01-0312-46   01-101  Ciudad Capital                Zona 1   

                                       ESTABLECIMIENTO  \
77   Instituto Guatemalteco De Educacion Radiofonic...   
209  Liceo De Bachillerato En Computacion Y Educaci...   
695     Escuela Privada De Educacion Fisica De Oriente   
809                           Liceo Tecnico Concepcion   
940                         Colegio Nacional Americano   

                       DIRECCION TELEFONO  \
77   No hay dirección registrada      NaN   
209  No hay dirección registrada      NaN   
695  No hay dirección registrada      NaN   
809  No hay dirección registrada      NaN   
940  No hay dirección registrada      NaN   

                                SUPERVISOR DIRECTOR          NIVEL   SECTOR  \
77           SONIA HAYDEE REYES MONTERROSO      NaN  DIVERSIFICADO  PRIVADO   
209                                    NaN      NaN  DIVERSIFICADO  PRIVADO   
695       ANGELA DEL ROSARIO GARCIA MARCOS      ---  DIVERSIFICADO  PRIVADO   
809                  LANDELINO CRUZ GARCIA      NaN  DIVERSIFICADO  PRIVADO   
940  MARIA ESTELA MARROQUIN CHUR DE ORTEGA      ---  DIVERSIFICADO  PRIVADO   

       AREA                 STATUS   MODALIDAD     JORNADA             PLAN  \
77    RURAL  CERRADA TEMPORALMENTE  MONOLINGUE    MATUTINA      A DISTANCIA   
209  URBANA  CERRADA TEMPORALMENTE  MONOLINGUE    MATUTINA  DIARIO(REGULAR)   
695  URBANA  CERRADA TEMPORALMENTE  MONOLINGUE  VESPERTINA  DIARIO(REGULAR)   
809  URBANA  CERRADA TEMPORALMENTE  MONOLINGUE  VESPERTINA  DIARIO(REGULAR)   
940  URBANA       TEMPORAL TITULOS  MONOLINGUE    MATUTINA  DIARIO(REGULAR)   

       DEPARTAMENTAL  
77      ALTA VERAPAZ  
209     ALTA VERAPAZ  
695       CHIQUIMULA  
809       CHIQUIMULA  
940  GUATEMALA NORTE

#### Conteo de establecimientos por dirección

In [40]:
df.DIRECCION.value_counts()

Cabecera Municipal                                                 242
Barrio El Centro                                                    63
No hay dirección registrada                                         55
Barrio El Porvenir                                                  18
Barrio El Calvario                                                  14
                                                                  ... 
1A. Calle 2-35 Zona 4 De Mixco Colonia Monte Verde                   1
Boulevard Sur Manzana C, Lote 22, Valle Dorado, San Cristobal        1
6A Calle 14-39, Aldea Sacoj Chiquito, Zona 6                         1
4A. Calle "A" 13-60, Sector B-2, Zona 8, Ciudad San Cristobal I      1
10A. Avenida 10-55, Zona 2                                           1
Name: DIRECCION, Length: 5108, dtype: int64

## Análisis para variable teléfono

#### Revisión de Formato

In [41]:
# Formato de datos
display(df.TELEFONO.unique().tolist()[0:10])


['77945104',
 '77367402',
 '49532425',
 '79514215',
 '79521468',
 '57101061',
 '54641454',
 nan,
 '79522555',
 '79945251']

#### Revisión de Nulls

In [42]:
display(df.loc[df.TELEFONO.isna(),:].head())
a=len(df.loc[df.TELEFONO.isna(),:])
b=len(df)
print('Cantidad de establecimientos sin dirección: '+str(a)+' siendo el '+str(round((a/b)*100,2))+' % de toda la base de datos')

CODIGO DISTRITO  DEPARTAMENTO MUNICIPIO  \
8   16-01-0150-46   16-006  Alta Verapaz     Coban   
49  16-01-1172-46   16-006  Alta Verapaz     Coban   
50  16-01-1173-46   16-006  Alta Verapaz     Coban   
66  16-01-2171-46   16-031  Alta Verapaz     Coban   
68  16-01-2947-46       65  Alta Verapaz     Coban   

                                      ESTABLECIMIENTO  \
8    Instituto Intercultrual Altaverapacencese -Iiav-   
49                  Colegio Particular Mixto Carlos V   
50                  Colegio Particular Mixto Carlos V   
66                  Instituto Mixto Nuevos Horizontes   
68  Instituto Privado Basico Y Bachillerato Por Ma...   

                                     DIRECCION TELEFONO  \
8                     3A. Avaenida 1-23 Zona 4      NaN   
49  6A. Calle 2-06 Zona 8 Periferico Sur Coban      NaN   
50  6A. Calle 2-06 Zona 8 Periferico Sur Coban      NaN   
66                    6A. Avenida 1-15, Zona 4      NaN   
68                       5A. Calle 1-98 Zona 3      NaN   

                       SUPERVISOR                            DIRECTOR  \
8                 EFRAIN CAAL CUC  GUILLERMO ESTUARDO VASQUEZ MORALES   
49                EFRAIN CAAL CUC                                 NaN   
50                EFRAIN CAAL CUC                                 NaN   
66  CESAR ROLANDO CHOCOOJ YALIBAT                                 NaN   
68                            NaN                                 NaN   

            NIVEL   SECTOR    AREA                 STATUS   MODALIDAD  \
8   DIVERSIFICADO  PRIVADO  URBANA  CERRADA TEMPORALMENTE    BILINGUE   
49  DIVERSIFICADO  PRIVADO  URBANA  CERRADA TEMPORALMENTE  MONOLINGUE   
50  DIVERSIFICADO  PRIVADO  URBANA  CERRADA TEMPORALMENTE  MONOLINGUE   
66  DIVERSIFICADO  PRIVADO  URBANA  CERRADA TEMPORALMENTE  MONOLINGUE   
68  DIVERSIFICADO  PRIVADO  URBANA  CERRADA TEMPORALMENTE  MONOLINGUE   

       JORNADA             PLAN DEPARTAMENTAL  
8        DOBLE    FIN DE SEMANA  ALTA VERAPAZ  
49    MATUTINA    FIN DE SEMANA  ALTA VERAPAZ  
50    MATUTINA  DIARIO(REGULAR)  ALTA VERAPAZ  
66    MATUTINA    FIN DE SEMANA  ALTA VERAPAZ  
68  VESPERTINA  DIARIO(REGULAR)  ALTA VERAPAZ

Cantidad de establecimientos sin dirección: 571 siendo el 7.17 % de toda la base de datos


In [43]:
df.loc[df.TELEFONO.isna(),'TELEFONO']='No hay teléfono registrado'

display(df.loc[df.TELEFONO=='No hay teléfono registrado',:].head())

CODIGO DISTRITO  DEPARTAMENTO MUNICIPIO  \
8   16-01-0150-46   16-006  Alta Verapaz     Coban   
49  16-01-1172-46   16-006  Alta Verapaz     Coban   
50  16-01-1173-46   16-006  Alta Verapaz     Coban   
66  16-01-2171-46   16-031  Alta Verapaz     Coban   
68  16-01-2947-46       65  Alta Verapaz     Coban   

                                      ESTABLECIMIENTO  \
8    Instituto Intercultrual Altaverapacencese -Iiav-   
49                  Colegio Particular Mixto Carlos V   
50                  Colegio Particular Mixto Carlos V   
66                  Instituto Mixto Nuevos Horizontes   
68  Instituto Privado Basico Y Bachillerato Por Ma...   

                                     DIRECCION                    TELEFONO  \
8                     3A. Avaenida 1-23 Zona 4  No hay teléfono registrado   
49  6A. Calle 2-06 Zona 8 Periferico Sur Coban  No hay teléfono registrado   
50  6A. Calle 2-06 Zona 8 Periferico Sur Coban  No hay teléfono registrado   
66                    6A. Avenida 1-15, Zona 4  No hay teléfono registrado   
68                       5A. Calle 1-98 Zona 3  No hay teléfono registrado   

                       SUPERVISOR                            DIRECTOR  \
8                 EFRAIN CAAL CUC  GUILLERMO ESTUARDO VASQUEZ MORALES   
49                EFRAIN CAAL CUC                                 NaN   
50                EFRAIN CAAL CUC                                 NaN   
66  CESAR ROLANDO CHOCOOJ YALIBAT                                 NaN   
68                            NaN                                 NaN   

            NIVEL   SECTOR    AREA                 STATUS   MODALIDAD  \
8   DIVERSIFICADO  PRIVADO  URBANA  CERRADA TEMPORALMENTE    BILINGUE   
49  DIVERSIFICADO  PRIVADO  URBANA  CERRADA TEMPORALMENTE  MONOLINGUE   
50  DIVERSIFICADO  PRIVADO  URBANA  CERRADA TEMPORALMENTE  MONOLINGUE   
66  DIVERSIFICADO  PRIVADO  URBANA  CERRADA TEMPORALMENTE  MONOLINGUE   
68  DIVERSIFICADO  PRIVADO  URBANA  CERRADA TEMPORALMENTE  MONOLINGUE   

       JORNADA             PLAN DEPARTAMENTAL  
8        DOBLE    FIN DE SEMANA  ALTA VERAPAZ  
49    MATUTINA    FIN DE SEMANA  ALTA VERAPAZ  
50    MATUTINA  DIARIO(REGULAR)  ALTA VERAPAZ  
66    MATUTINA    FIN DE SEMANA  ALTA VERAPAZ  
68  VESPERTINA  DIARIO(REGULAR)  ALTA VERAPAZ

#### Convertir a String

In [44]:
df.TELEFONO=df.TELEFONO.astype(str)
df.TELEFONO

0         77945104
1         77367402
2         49532425
3         79514215
4         79521468
           ...    
7959    79418369.0
7960    79418369.0
7961    31863599.0
7962    55958103.0
7963    41447589.0
Name: TELEFONO, Length: 7964, dtype: object

#### Se elimina el '.0' y cualquier otro decimal si fuese mal escrito o detectado como un número, utilizando split y tomando todo lo que está al lado izquierdo del punto

In [45]:
df.TELEFONO=df.TELEFONO.str.split('.',expand=True)[0]
df.TELEFONO

0       77945104
1       77367402
2       49532425
3       79514215
4       79521468
          ...   
7959    79418369
7960    79418369
7961    31863599
7962    55958103
7963    41447589
Name: TELEFONO, Length: 7964, dtype: object

#### Eliminación de espacios en blanco no necesarios

In [46]:
df.TELEFONO=df.TELEFONO.str.strip()
df

CODIGO DISTRITO  DEPARTAMENTO  MUNICIPIO  \
0     16-01-0138-46   16-031  Alta Verapaz      Coban   
1     16-01-0139-46   16-031  Alta Verapaz      Coban   
2     16-01-0140-46   16-031  Alta Verapaz      Coban   
3     16-01-0141-46   16-005  Alta Verapaz      Coban   
4     16-01-0142-46   16-005  Alta Verapaz      Coban   
...             ...      ...           ...        ...   
7959  19-09-0040-46   19-026        Zacapa   La Union   
7960  19-09-0902-46   19-026        Zacapa   La Union   
7961  19-10-0013-46   19-016        Zacapa      Huite   
7962  19-10-1009-46   19-016        Zacapa      Huite   
7963  19-11-0018-46   19-020        Zacapa  San Jorge   

                                        ESTABLECIMIENTO  \
0                                         Colegio Coban   
1                      Colegio Particular Mixto Verapaz   
2                               Colegio "La Inmaculada"   
3              Escuela Nacional De Ciencias Comerciales   
4     Instituto Normal Mixto Del Norte 'Emilio Rosal...   
...                                                 ...   
7959                       Liceo Particular Mixto Jireh   
7960         Liceo Particular Mixto Jireh (Por Madurez)   
7961                            Instituto Diversificado   
7962  Instituto Mixto De Educacion Diversificada Por...   
7963  Instituto Mixto De Educacion Diversificada Por...   

                                   DIRECCION  TELEFONO  \
0     Km.2 Salida A San Juan Chamelco Zona 8  77945104   
1               Km 209.5 Entrada A La Ciudad  77367402   
2                  7A. Avenida 11-109 Zona 6  49532425   
3                      2A Calle 11-10 Zona 2  79514215   
4                        3A Ave 6-23 Zona 11  79521468   
...                                      ...       ...   
7959                            Barrio Nuevo  79418369   
7960                            Barrio Nuevo  79418369   
7961                     Barrio Buenos Aires  31863599   
7962                         Barrio El Campo  55958103   
7963                        Barrio El Centro  41447589   

                               SUPERVISOR  \
0           CESAR ROLANDO CHOCOOJ YALIBAT   
1           CESAR ROLANDO CHOCOOJ YALIBAT   
2           CESAR ROLANDO CHOCOOJ YALIBAT   
3                 JUAN CARLOS WINTER CAAL   
4                 JUAN CARLOS WINTER CAAL   
...                                   ...   
7959  BERTA ALICIA LEIVA CORDON DE GARCIA   
7960  BERTA ALICIA LEIVA CORDON DE GARCIA   
7961         WENDER HUMBERTO MATEO NAJERA   
7962         WENDER HUMBERTO MATEO NAJERA   
7963                      ALBA LUZ MENDEZ   

                                 DIRECTOR          NIVEL       SECTOR    AREA  \
0               JULIO CESAR VILLELA AMADO  DIVERSIFICADO      PRIVADO  URBANA   
1                                     NaN  DIVERSIFICADO      PRIVADO  URBANA   
2                  MERCEDES QUIROS QUIROS  DIVERSIFICADO      PRIVADO  URBANA   
3                                     NaN  DIVERSIFICADO      OFICIAL  URBANA   
4                                     NaN  DIVERSIFICADO      OFICIAL  URBANA   
...                                   ...            ...          ...     ...   
7959  ANA MARÍA CUELLAR GUERRA DE RAMÍREZ  DIVERSIFICADO      PRIVADO  URBANA   
7960  ANA MARÍA CUELLAR GUERRA DE RAMÍREZ  DIVERSIFICADO      PRIVADO  URBANA   
7961            ERIC MANUEL PÉREZ AGUILAR  DIVERSIFICADO      OFICIAL  URBANA   
7962            ROBIDIO PORTILLO SALGUERO  DIVERSIFICADO  COOPERATIVA  URBANA   
7963            VICTOR HUGO GUERRA MONROY  DIVERSIFICADO  COOPERATIVA  URBANA   

       STATUS   MODALIDAD     JORNADA             PLAN DEPARTAMENTAL  
0     ABIERTA  MONOLINGUE    MATUTINA  DIARIO(REGULAR)  ALTA VERAPAZ  
1     ABIERTA  MONOLINGUE    MATUTINA  DIARIO(REGULAR)  ALTA VERAPAZ  
2     ABIERTA  MONOLINGUE    MATUTINA  DIARIO(REGULAR)  ALTA VERAPAZ  
3     ABIERTA  MONOLINGUE    MATUTINA  DIARIO(REGULAR)  ALTA VERAPAZ  
4     ABIERTA    BILINGUE  VESPERTINA  DIARIO(REGULAR

#### Debido a que existén establecimientos con más de un número telefónico vamos a habilitar las columnas auxiliares TELEFONO 2 y TELEFONO 3

In [47]:
telefonos=df.TELEFONO.str.split('-',expand=True)

df['TELEFONO']=telefonos[0]
df['TELEFONO 2']=telefonos[1]
df['TELEFONO 3']=telefonos[2]



In [48]:
df

CODIGO DISTRITO  DEPARTAMENTO  MUNICIPIO  \
0     16-01-0138-46   16-031  Alta Verapaz      Coban   
1     16-01-0139-46   16-031  Alta Verapaz      Coban   
2     16-01-0140-46   16-031  Alta Verapaz      Coban   
3     16-01-0141-46   16-005  Alta Verapaz      Coban   
4     16-01-0142-46   16-005  Alta Verapaz      Coban   
...             ...      ...           ...        ...   
7959  19-09-0040-46   19-026        Zacapa   La Union   
7960  19-09-0902-46   19-026        Zacapa   La Union   
7961  19-10-0013-46   19-016        Zacapa      Huite   
7962  19-10-1009-46   19-016        Zacapa      Huite   
7963  19-11-0018-46   19-020        Zacapa  San Jorge   

                                        ESTABLECIMIENTO  \
0                                         Colegio Coban   
1                      Colegio Particular Mixto Verapaz   
2                               Colegio "La Inmaculada"   
3              Escuela Nacional De Ciencias Comerciales   
4     Instituto Normal Mixto Del Norte 'Emilio Rosal...   
...                                                 ...   
7959                       Liceo Particular Mixto Jireh   
7960         Liceo Particular Mixto Jireh (Por Madurez)   
7961                            Instituto Diversificado   
7962  Instituto Mixto De Educacion Diversificada Por...   
7963  Instituto Mixto De Educacion Diversificada Por...   

                                   DIRECCION  TELEFONO  \
0     Km.2 Salida A San Juan Chamelco Zona 8  77945104   
1               Km 209.5 Entrada A La Ciudad  77367402   
2                  7A. Avenida 11-109 Zona 6  49532425   
3                      2A Calle 11-10 Zona 2  79514215   
4                        3A Ave 6-23 Zona 11  79521468   
...                                      ...       ...   
7959                            Barrio Nuevo  79418369   
7960                            Barrio Nuevo  79418369   
7961                     Barrio Buenos Aires  31863599   
7962                         Barrio El Campo  55958103   
7963                        Barrio El Centro  41447589   

                               SUPERVISOR  \
0           CESAR ROLANDO CHOCOOJ YALIBAT   
1           CESAR ROLANDO CHOCOOJ YALIBAT   
2           CESAR ROLANDO CHOCOOJ YALIBAT   
3                 JUAN CARLOS WINTER CAAL   
4                 JUAN CARLOS WINTER CAAL   
...                                   ...   
7959  BERTA ALICIA LEIVA CORDON DE GARCIA   
7960  BERTA ALICIA LEIVA CORDON DE GARCIA   
7961         WENDER HUMBERTO MATEO NAJERA   
7962         WENDER HUMBERTO MATEO NAJERA   
7963                      ALBA LUZ MENDEZ   

                                 DIRECTOR          NIVEL       SECTOR    AREA  \
0               JULIO CESAR VILLELA AMADO  DIVERSIFICADO      PRIVADO  URBANA   
1                                     NaN  DIVERSIFICADO      PRIVADO  URBANA   
2                  MERCEDES QUIROS QUIROS  DIVERSIFICADO      PRIVADO  URBANA   
3                                     NaN  DIVERSIFICADO      OFICIAL  URBANA   
4                                     NaN  DIVERSIFICADO      OFICIAL  URBANA   
...                                   ...            ...          ...     ...   
7959  ANA MARÍA CUELLAR GUERRA DE RAMÍREZ  DIVERSIFICADO      PRIVADO  URBANA   
7960  ANA MARÍA CUELLAR GUERRA DE RAMÍREZ  DIVERSIFICADO      PRIVADO  URBANA   
7961            ERIC MANUEL PÉREZ AGUILAR  DIVERSIFICADO      OFICIAL  URBANA   
7962            ROBIDIO PORTILLO SALGUERO  DIVERSIFICADO  COOPERATIVA  URBANA   
7963            VICTOR HUGO GUERRA MONROY  DIVERSIFICADO  COOPERATIVA  URBANA   

       STATUS   MODALIDAD     JORNADA             PLAN DEPARTAMENTAL  \
0     ABIERTA  MONOLINGUE    MATUTINA  DIARIO(REGULAR)  ALTA VERAPAZ   
1     ABIERTA  MONOLINGUE    MATUTINA  DIARIO(REGULAR)  ALTA VERAPAZ   
2     ABIERTA  MONOLINGUE    MATUTINA  DIARIO(REGULAR)  ALTA VERAPAZ   
3     ABIERTA  MONOLINGUE    MATUTINA  DIARIO(REGULAR)  ALTA VERAPAZ   
4     ABIERTA    BILINGUE  VESPERTINA  DIARIO(RE

#### Función para corregir números de teléfono

In [49]:
def corregir_telefono(i,a):
    
    if(i==None):
        return 'No hay teléfono registrado'
    
    
    if(len(i)<8):
        if(a==None):
            i='No hay teléfono registrado'
        
        elif((len(i)==4)&(len(a)==4)):
            i=i+a
            
        else:
            i='No hay teléfono registrado'
        
    
    elif (len(i)==8):
        return i
    
    else:
        return 'No hay teléfono registrado'


df.TELEFONO=df.apply(lambda row : corregir_telefono(row['TELEFONO'],row['TELEFONO 2']), axis = 1)

df['TELEFONO 2']=df.apply(lambda row : corregir_telefono(row['TELEFONO 2'],row['TELEFONO 3']), axis = 1)


del df['TELEFONO 3']

# try:
    
# except:
#     print('Todos los teléfonos ya han sido restaurados')
    

#### Conteo de establecimientos por dirección

In [50]:
df.TELEFONO.value_counts()

No hay teléfono registrado    601
77602663                       12
79480009                       12
22093200                       10
77746400                       10
                             ... 
66296659                        1
47115341                        1
54133448                        1
66591440                        1
41447589                        1
Name: TELEFONO, Length: 4616, dtype: int64

## Análisis para variable Supervisor

#### Revisión de Formato

In [51]:
# Formato de datos
display(df.SUPERVISOR.unique().tolist()[0:10])


['CESAR ROLANDO CHOCOOJ YALIBAT',
 'JUAN CARLOS WINTER CAAL',
 'EFRAIN CAAL CUC',
 'JUVENTINA ALVARADO PEREZ',
 'MAXIMILIANO CHUB ICAL',
 nan,
 'SONIA HAYDEE REYES MONTERROSO',
 'OSCAR RENE XONA MORAN',
 'ODILIA VANESSA MAAS CHUN DE MUS',
 'ROBERTO RAX']

#### Revisión de Nulls

In [52]:
display(df.loc[df.SUPERVISOR.isna(),:].head())
a=len(df.loc[df.SUPERVISOR.isna(),:])
b=len(df)
print('Cantidad de establecimientos sin dato de supervisor: '+str(a)+' siendo el '+str(round((a/b)*100,2))+' % de toda la base de datos')

CODIGO DISTRITO   DEPARTAMENTO          MUNICIPIO  \
68   16-01-2947-46       65   Alta Verapaz              Coban   
209  16-10-3816-46       14   Alta Verapaz  San Juan Chamelco   
344  15-03-0134-46       21   Baja Verapaz            Rabinal   
637  04-11-0995-46        4  Chimaltenango         Acatenango   
674  04-15-0599-46       15  Chimaltenango           Zaragoza   

                                       ESTABLECIMIENTO  \
68   Instituto Privado Basico Y Bachillerato Por Ma...   
209  Liceo De Bachillerato En Computacion Y Educaci...   
344  Liceo Privado Mixto De Ciencias Comerciales 'R...   
637              Instituto Practico De Educacion Media   
674  Instituto Privado Mixto De Educacion Diversifi...   

                       DIRECCION                    TELEFONO SUPERVISOR  \
68         5A. Calle 1-98 Zona 3  No hay teléfono registrado        NaN   
209  No hay dirección registrada  No hay teléfono registrado        NaN   
344      3A. Avenida 1-55 Zona 3  No hay teléfono registrado        NaN   
637               Canton Tercero  No hay teléfono registrado        NaN   
674       6A. Avenida Norte No.2  No hay teléfono registrado        NaN   

    DIRECTOR          NIVEL       SECTOR    AREA                 STATUS  \
68       NaN  DIVERSIFICADO      PRIVADO  URBANA  CERRADA TEMPORALMENTE   
209      NaN  DIVERSIFICADO      PRIVADO  URBANA  CERRADA TEMPORALMENTE   
344      NaN  DIVERSIFICADO      PRIVADO  URBANA  CERRADA TEMPORALMENTE   
637      NaN  DIVERSIFICADO  COOPERATIVA   RURAL  CERRADA TEMPORALMENTE   
674      NaN  DIVERSIFICADO      PRIVADO  URBANA  CERRADA TEMPORALMENTE   

      MODALIDAD     JORNADA             PLAN  DEPARTAMENTAL  \
68   MONOLINGUE  VESPERTINA  DIARIO(REGULAR)   ALTA VERAPAZ   
209  MONOLINGUE    MATUTINA  DIARIO(REGULAR)   ALTA VERAPAZ   
344  MONOLINGUE  VESPERTINA  DIARIO(REGULAR)   BAJA VERAPAZ   
637  MONOLINGUE    NOCTURNA  DIARIO(REGULAR)  CHIMALTENANGO   
674  MONOLINGUE    NOCTURNA  DIARIO(REGULAR)  CHIMALTENANGO   

                     TELEFONO 2  
68   No hay teléfono registrado  
209  No hay teléfono registrado  
344  No hay teléfono registrado  
637  No hay teléfono registrado  
674  No hay teléfono registrado

Cantidad de establecimientos sin dato de supervisor: 213 siendo el 2.67 % de toda la base de datos


#### Formato a Proper Case 

In [53]:
df.SUPERVISOR=df.SUPERVISOR.str.title()

#### Eliminación de espacios en blanco no necesarios

In [54]:
df.SUPERVISOR=df.SUPERVISOR.str.strip()

#### Eliminación de acentos para estandarización

In [55]:
df.SUPERVISOR=df.SUPERVISOR.str.replace('á','a')
df.SUPERVISOR=df.SUPERVISOR.str.replace('é','e')
df.SUPERVISOR=df.SUPERVISOR.str.replace('í','i')
df.SUPERVISOR=df.SUPERVISOR.str.replace('ó','o')
df.SUPERVISOR=df.SUPERVISOR.str.replace('ú','u')

#### Se agrega 'No hay dirección registrada' a los datos faltantes

In [56]:
df.loc[df.SUPERVISOR.isna(),'SUPERVISOR']='No hay supervisor registrado'

display(df.loc[df.SUPERVISOR=='No hay supervisor registrado',:].head())

CODIGO DISTRITO   DEPARTAMENTO          MUNICIPIO  \
68   16-01-2947-46       65   Alta Verapaz              Coban   
209  16-10-3816-46       14   Alta Verapaz  San Juan Chamelco   
344  15-03-0134-46       21   Baja Verapaz            Rabinal   
637  04-11-0995-46        4  Chimaltenango         Acatenango   
674  04-15-0599-46       15  Chimaltenango           Zaragoza   

                                       ESTABLECIMIENTO  \
68   Instituto Privado Basico Y Bachillerato Por Ma...   
209  Liceo De Bachillerato En Computacion Y Educaci...   
344  Liceo Privado Mixto De Ciencias Comerciales 'R...   
637              Instituto Practico De Educacion Media   
674  Instituto Privado Mixto De Educacion Diversifi...   

                       DIRECCION                    TELEFONO  \
68         5A. Calle 1-98 Zona 3  No hay teléfono registrado   
209  No hay dirección registrada  No hay teléfono registrado   
344      3A. Avenida 1-55 Zona 3  No hay teléfono registrado   
637               Canton Tercero  No hay teléfono registrado   
674       6A. Avenida Norte No.2  No hay teléfono registrado   

                       SUPERVISOR DIRECTOR          NIVEL       SECTOR  \
68   No hay supervisor registrado      NaN  DIVERSIFICADO      PRIVADO   
209  No hay supervisor registrado      NaN  DIVERSIFICADO      PRIVADO   
344  No hay supervisor registrado      NaN  DIVERSIFICADO      PRIVADO   
637  No hay supervisor registrado      NaN  DIVERSIFICADO  COOPERATIVA   
674  No hay supervisor registrado      NaN  DIVERSIFICADO      PRIVADO   

       AREA                 STATUS   MODALIDAD     JORNADA             PLAN  \
68   URBANA  CERRADA TEMPORALMENTE  MONOLINGUE  VESPERTINA  DIARIO(REGULAR)   
209  URBANA  CERRADA TEMPORALMENTE  MONOLINGUE    MATUTINA  DIARIO(REGULAR)   
344  URBANA  CERRADA TEMPORALMENTE  MONOLINGUE  VESPERTINA  DIARIO(REGULAR)   
637   RURAL  CERRADA TEMPORALMENTE  MONOLINGUE    NOCTURNA  DIARIO(REGULAR)   
674  URBANA  CERRADA TEMPORALMENTE  MONOLINGUE    NOCTURNA  DIARIO(REGULAR)   

     DEPARTAMENTAL                  TELEFONO 2  
68    ALTA VERAPAZ  No hay teléfono registrado  
209   ALTA VERAPAZ  No hay teléfono registrado  
344   BAJA VERAPAZ  No hay teléfono registrado  
637  CHIMALTENANGO  No hay teléfono registrado  
674  CHIMALTENANGO  No hay teléfono registrado

#### Conteo de supervisores por dirección

In [57]:
df.SUPERVISOR.value_counts()

Carlos Humberto Gonzalez De Leon    257
No hay supervisor registrado        213
Miguel Angel Armas Rocha            176
Remy Arturo Sinay Gudiel            137
Juan Enrique Martinez Solano        128
                                   ... 
Maria Del Carmen Sandoval Lopez       1
Santiago Cruz Serrano                 1
Jose Francisco Olan Ponce             1
Veronica Esperanza Aguilar Gomez      1
Alba Luz Mendez                       1
Name: SUPERVISOR, Length: 638, dtype: int64

## Análisis para variable Director

#### Revisión de Formato

In [58]:
# Formato de datos
display(df.DIRECTOR.unique().tolist()[0:10])


['JULIO CESAR VILLELA AMADO',
 nan,
 'MERCEDES QUIROS QUIROS',
 'HECOTR WALDEMAR TOT COY',
 'LUIS FERNANDO SOTO',
 'GUILLERMO ESTUARDO VASQUEZ MORALES',
 'JORGE BENEDICTO COC POP',
 'GEHU ABRAHAM CAAL MAAS',
 'MARIA MAGDALENA BOL CU',
 'ELIGIO PASTOR GARCÍA']

#### Revisión de Nulls

In [59]:
display(df.loc[df.DIRECTOR.isna(),:].head())
a=len(df.loc[df.DIRECTOR.isna(),:])
b=len(df)
print('Cantidad de establecimientos sin director: '+str(a)+' siendo el '+str(round((a/b)*100,2))+' % de toda la base de datos')

CODIGO DISTRITO  DEPARTAMENTO MUNICIPIO  \
1   16-01-0139-46   16-031  Alta Verapaz     Coban   
3   16-01-0141-46   16-005  Alta Verapaz     Coban   
4   16-01-0142-46   16-005  Alta Verapaz     Coban   
49  16-01-1172-46   16-006  Alta Verapaz     Coban   
50  16-01-1173-46   16-006  Alta Verapaz     Coban   

                                      ESTABLECIMIENTO  \
1                    Colegio Particular Mixto Verapaz   
3            Escuela Nacional De Ciencias Comerciales   
4   Instituto Normal Mixto Del Norte 'Emilio Rosal...   
49                  Colegio Particular Mixto Carlos V   
50                  Colegio Particular Mixto Carlos V   

                                     DIRECCION                    TELEFONO  \
1                 Km 209.5 Entrada A La Ciudad                    77367402   
3                        2A Calle 11-10 Zona 2                    79514215   
4                          3A Ave 6-23 Zona 11                    79521468   
49  6A. Calle 2-06 Zona 8 Periferico Sur Coban  No hay teléfono registrado   
50  6A. Calle 2-06 Zona 8 Periferico Sur Coban  No hay teléfono registrado   

                       SUPERVISOR DIRECTOR          NIVEL   SECTOR    AREA  \
1   Cesar Rolando Chocooj Yalibat      NaN  DIVERSIFICADO  PRIVADO  URBANA   
3         Juan Carlos Winter Caal      NaN  DIVERSIFICADO  OFICIAL  URBANA   
4         Juan Carlos Winter Caal      NaN  DIVERSIFICADO  OFICIAL  URBANA   
49                Efrain Caal Cuc      NaN  DIVERSIFICADO  PRIVADO  URBANA   
50                Efrain Caal Cuc      NaN  DIVERSIFICADO  PRIVADO  URBANA   

                   STATUS   MODALIDAD     JORNADA             PLAN  \
1                 ABIERTA  MONOLINGUE    MATUTINA  DIARIO(REGULAR)   
3                 ABIERTA  MONOLINGUE    MATUTINA  DIARIO(REGULAR)   
4                 ABIERTA    BILINGUE  VESPERTINA  DIARIO(REGULAR)   
49  CERRADA TEMPORALMENTE  MONOLINGUE    MATUTINA    FIN DE SEMANA   
50  CERRADA TEMPORALMENTE  MONOLINGUE    MATUTINA  DIARIO(REGULAR)   

   DEPARTAMENTAL                  TELEFONO 2  
1   ALTA VERAPAZ  No hay teléfono registrado  
3   ALTA VERAPAZ  No hay teléfono registrado  
4   ALTA VERAPAZ  No hay teléfono registrado  
49  ALTA VERAPAZ  No hay teléfono registrado  
50  ALTA VERAPAZ  No hay teléfono registrado

Cantidad de establecimientos sin director: 1653 siendo el 20.76 % de toda la base de datos


#### Formato a Proper Case 

In [60]:
df.DIRECTOR=df.DIRECTOR.str.title()

#### Eliminación de espacios en blanco no necesarios

In [61]:
df.DIRECTOR=df.DIRECTOR.str.strip()

#### Eliminación de acentos para estandarización

In [62]:
df.DIRECTOR=df.DIRECTOR.str.replace('á','a')
df.DIRECTOR=df.DIRECTOR.str.replace('é','e')
df.DIRECTOR=df.DIRECTOR.str.replace('í','i')
df.DIRECTOR=df.DIRECTOR.str.replace('ó','o')
df.DIRECTOR=df.DIRECTOR.str.replace('ú','u')

#### Se agrega 'No hay dirección registrada' a los datos faltantes

In [63]:
df.loc[df.DIRECTOR.isna(),'DIRECTOR']='No hay director registrado'
df.loc[df.DIRECTOR.str.contains('-'),'DIRECTOR']='No hay director registrado'
df.loc[df.DIRECTOR=='.','DIRECTOR']='No hay director registrado'


display(df.loc[df.DIRECTOR=='No hay director registrado',:].head())

CODIGO DISTRITO  DEPARTAMENTO MUNICIPIO  \
1   16-01-0139-46   16-031  Alta Verapaz     Coban   
3   16-01-0141-46   16-005  Alta Verapaz     Coban   
4   16-01-0142-46   16-005  Alta Verapaz     Coban   
16  16-01-0558-46   16-005  Alta Verapaz     Coban   
17  16-01-0559-46   16-005  Alta Verapaz     Coban   

                                      ESTABLECIMIENTO  \
1                    Colegio Particular Mixto Verapaz   
3            Escuela Nacional De Ciencias Comerciales   
4   Instituto Normal Mixto Del Norte 'Emilio Rosal...   
16  Instituto Normal Preprimaria Bilingue Adscrito...   
17  Instituto Normal Primaria Bilingue Intercultur...   

                       DIRECCION  TELEFONO                     SUPERVISOR  \
1   Km 209.5 Entrada A La Ciudad  77367402  Cesar Rolando Chocooj Yalibat   
3          2A Calle 11-10 Zona 2  79514215        Juan Carlos Winter Caal   
4            3A Ave 6-23 Zona 11  79521468        Juan Carlos Winter Caal   
16         3A. Ave. 6-23 Zona 11  79521468        Juan Carlos Winter Caal   
17         3A. Ave. 6-23 Zona 11  79521468        Juan Carlos Winter Caal   

                      DIRECTOR          NIVEL   SECTOR    AREA   STATUS  \
1   No hay director registrado  DIVERSIFICADO  PRIVADO  URBANA  ABIERTA   
3   No hay director registrado  DIVERSIFICADO  OFICIAL  URBANA  ABIERTA   
4   No hay director registrado  DIVERSIFICADO  OFICIAL  URBANA  ABIERTA   
16  No hay director registrado  DIVERSIFICADO  OFICIAL  URBANA  ABIERTA   
17  No hay director registrado  DIVERSIFICADO  OFICIAL  URBANA  ABIERTA   

     MODALIDAD     JORNADA             PLAN DEPARTAMENTAL  \
1   MONOLINGUE    MATUTINA  DIARIO(REGULAR)  ALTA VERAPAZ   
3   MONOLINGUE    MATUTINA  DIARIO(REGULAR)  ALTA VERAPAZ   
4     BILINGUE  VESPERTINA  DIARIO(REGULAR)  ALTA VERAPAZ   
16  MONOLINGUE  VESPERTINA  DIARIO(REGULAR)  ALTA VERAPAZ   
17  MONOLINGUE  VESPERTINA  DIARIO(REGULAR)  ALTA VERAPAZ   

                    TELEFONO 2  
1   No hay teléfono registrado  
3   No hay teléfono registrado  
4   No hay teléfono registrado  
16  No hay teléfono registrado  
17  No hay teléfono registrado

#### Conteo de establecimientos por director

In [64]:
df.DIRECTOR.value_counts()

No hay director registrado                  2046
Maria Dolores Perez Tuchan                    11
Jorge Granados Guzman                          9
Maria Del Carmen Aviles Corado                 8
Demecio Herculano Orozco Gonzalez              8
                                            ... 
Jose Lus Fernandez Romano                      1
Claudia Roselia Cifuentes Alas                 1
Pem Maria Elisa Flores Herrera De Franco       1
Juana Patricia Echeverria Pech                 1
Robidio Portillo Salguero                      1
Name: DIRECTOR, Length: 3876, dtype: int64

## Análisis para variable Nivel

No hay necesidad de transformaciones, solo se coloca el title case

#### Revisión de Nulls

In [65]:
display(df.loc[df.NIVEL.isna(),:].head())
a=len(df.loc[df.NIVEL.isna(),:])
b=len(df)
print('Cantidad de establecimientos sin NIVEL: '+str(a)+' siendo el '+str(round((a/b)*100,2))+' % de toda la base de datos')

Empty DataFrame
Columns: [CODIGO, DISTRITO, DEPARTAMENTO, MUNICIPIO, ESTABLECIMIENTO, DIRECCION, TELEFONO, SUPERVISOR, DIRECTOR, NIVEL, SECTOR, AREA, STATUS, MODALIDAD, JORNADA, PLAN, DEPARTAMENTAL, TELEFONO 2]
Index: []

Cantidad de establecimientos sin NIVEL: 0 siendo el 0.0 % de toda la base de datos


#### Conteo de variantes

In [66]:
df.NIVEL.value_counts()

DIVERSIFICADO    7964
Name: NIVEL, dtype: int64

In [67]:
df.NIVEL=df.NIVEL.str.title()
df.NIVEL

0       Diversificado
1       Diversificado
2       Diversificado
3       Diversificado
4       Diversificado
            ...      
7959    Diversificado
7960    Diversificado
7961    Diversificado
7962    Diversificado
7963    Diversificado
Name: NIVEL, Length: 7964, dtype: object

## Análisis para variable Sector
No hay necesidad de transformaciones, solo se coloca el title case

#### Revisión de Nulls

In [68]:
display(df.loc[df.SECTOR.isna(),:].head())
a=len(df.loc[df.SECTOR.isna(),:])
b=len(df)
print('Cantidad de establecimientos sin SECTOR: '+str(a)+' siendo el '+str(round((a/b)*100,2))+' % de toda la base de datos')

Empty DataFrame
Columns: [CODIGO, DISTRITO, DEPARTAMENTO, MUNICIPIO, ESTABLECIMIENTO, DIRECCION, TELEFONO, SUPERVISOR, DIRECTOR, NIVEL, SECTOR, AREA, STATUS, MODALIDAD, JORNADA, PLAN, DEPARTAMENTAL, TELEFONO 2]
Index: []

Cantidad de establecimientos sin SECTOR: 0 siendo el 0.0 % de toda la base de datos


#### Conteo de variantes

In [69]:
df.SECTOR.value_counts()

PRIVADO        6759
OFICIAL         856
COOPERATIVA     227
MUNICIPAL       122
Name: SECTOR, dtype: int64

In [70]:
df.SECTOR=df.SECTOR.str.title()
df.SECTOR

0           Privado
1           Privado
2           Privado
3           Oficial
4           Oficial
           ...     
7959        Privado
7960        Privado
7961        Oficial
7962    Cooperativa
7963    Cooperativa
Name: SECTOR, Length: 7964, dtype: object

## Análisis para variable Área
No hay necesidad de transformaciones, solo se coloca el title case

#### Revisión de Nulls

In [71]:
display(df.loc[df.AREA.isna(),:].head())
a=len(df.loc[df.AREA.isna(),:])
b=len(df)
print('Cantidad de establecimientos sin AREA: '+str(a)+' siendo el '+str(round((a/b)*100,2))+' % de toda la base de datos')

Empty DataFrame
Columns: [CODIGO, DISTRITO, DEPARTAMENTO, MUNICIPIO, ESTABLECIMIENTO, DIRECCION, TELEFONO, SUPERVISOR, DIRECTOR, NIVEL, SECTOR, AREA, STATUS, MODALIDAD, JORNADA, PLAN, DEPARTAMENTAL, TELEFONO 2]
Index: []

Cantidad de establecimientos sin AREA: 0 siendo el 0.0 % de toda la base de datos


#### Conteo de variantes

In [72]:
df.AREA.value_counts()

URBANA             6511
RURAL              1452
SIN ESPECIFICAR       1
Name: AREA, dtype: int64

In [73]:
df.AREA=df.AREA.str.title()
df.AREA

0       Urbana
1       Urbana
2       Urbana
3       Urbana
4       Urbana
         ...  
7959    Urbana
7960    Urbana
7961    Urbana
7962    Urbana
7963    Urbana
Name: AREA, Length: 7964, dtype: object

## Análisis para variable Status
No hay necesidad de transformaciones, solo se coloca el title case

#### Revisión de Nulls

In [74]:
display(df.loc[df.STATUS.isna(),:].head())
a=len(df.loc[df.STATUS.isna(),:])
b=len(df)
print('Cantidad de establecimientos sin STATUS: '+str(a)+' siendo el '+str(round((a/b)*100,2))+' % de toda la base de datos')

Empty DataFrame
Columns: [CODIGO, DISTRITO, DEPARTAMENTO, MUNICIPIO, ESTABLECIMIENTO, DIRECCION, TELEFONO, SUPERVISOR, DIRECTOR, NIVEL, SECTOR, AREA, STATUS, MODALIDAD, JORNADA, PLAN, DEPARTAMENTAL, TELEFONO 2]
Index: []

Cantidad de establecimientos sin STATUS: 0 siendo el 0.0 % de toda la base de datos


#### Conteo de variantes

In [75]:
df.STATUS.value_counts()

ABIERTA                  5838
CERRADA TEMPORALMENTE    1995
TEMPORAL TITULOS          128
TEMPORAL NOMBRAMIENTO       3
Name: STATUS, dtype: int64

In [76]:
df.STATUS=df.STATUS.str.title()
df.STATUS

0       Abierta
1       Abierta
2       Abierta
3       Abierta
4       Abierta
         ...   
7959    Abierta
7960    Abierta
7961    Abierta
7962    Abierta
7963    Abierta
Name: STATUS, Length: 7964, dtype: object

## Análisis para variable Modalidad

#### Revisión de Nulls

In [77]:
display(df.loc[df.MODALIDAD.isna(),:].head())
a=len(df.loc[df.MODALIDAD.isna(),:])
b=len(df)
print('Cantidad de establecimientos sin MODALIDAD: '+str(a)+' siendo el '+str(round((a/b)*100,2))+' % de toda la base de datos')

Empty DataFrame
Columns: [CODIGO, DISTRITO, DEPARTAMENTO, MUNICIPIO, ESTABLECIMIENTO, DIRECCION, TELEFONO, SUPERVISOR, DIRECTOR, NIVEL, SECTOR, AREA, STATUS, MODALIDAD, JORNADA, PLAN, DEPARTAMENTAL, TELEFONO 2]
Index: []

Cantidad de establecimientos sin MODALIDAD: 0 siendo el 0.0 % de toda la base de datos


#### Conteo de variantes

In [78]:
df.MODALIDAD.value_counts()

MONOLINGUE    7698
BILINGUE       266
Name: MODALIDAD, dtype: int64

In [79]:
df.MODALIDAD=df.MODALIDAD.str.title()
df.MODALIDAD

0       Monolingue
1       Monolingue
2       Monolingue
3       Monolingue
4         Bilingue
           ...    
7959    Monolingue
7960    Monolingue
7961    Monolingue
7962    Monolingue
7963    Monolingue
Name: MODALIDAD, Length: 7964, dtype: object

## Análisis para variable Jornada

#### Revisión de Nulls

In [80]:
display(df.loc[df.JORNADA.isna(),:].head())
a=len(df.loc[df.JORNADA.isna(),:])
b=len(df)
print('Cantidad de establecimientos sin JORNADA: '+str(a)+' siendo el '+str(round((a/b)*100,2))+' % de toda la base de datos')

Empty DataFrame
Columns: [CODIGO, DISTRITO, DEPARTAMENTO, MUNICIPIO, ESTABLECIMIENTO, DIRECCION, TELEFONO, SUPERVISOR, DIRECTOR, NIVEL, SECTOR, AREA, STATUS, MODALIDAD, JORNADA, PLAN, DEPARTAMENTAL, TELEFONO 2]
Index: []

Cantidad de establecimientos sin JORNADA: 0 siendo el 0.0 % de toda la base de datos


#### Conteo de variantes

In [81]:
df.JORNADA.value_counts()

DOBLE         2999
VESPERTINA    2414
MATUTINA      2187
NOCTURNA       269
INTERMEDIA      95
Name: JORNADA, dtype: int64

In [82]:
df.JORNADA=df.JORNADA.str.title()
df.JORNADA

0         Matutina
1         Matutina
2         Matutina
3         Matutina
4       Vespertina
           ...    
7959      Matutina
7960         Doble
7961      Nocturna
7962    Vespertina
7963      Matutina
Name: JORNADA, Length: 7964, dtype: object

## Análisis para variable Plan

#### Revisión de Nulls

In [83]:
display(df.loc[df.PLAN.isna(),:].head())
a=len(df.loc[df.PLAN.isna(),:])
b=len(df)
print('Cantidad de establecimientos sin PLAN: '+str(a)+' siendo el '+str(round((a/b)*100,2))+' % de toda la base de datos')

Empty DataFrame
Columns: [CODIGO, DISTRITO, DEPARTAMENTO, MUNICIPIO, ESTABLECIMIENTO, DIRECCION, TELEFONO, SUPERVISOR, DIRECTOR, NIVEL, SECTOR, AREA, STATUS, MODALIDAD, JORNADA, PLAN, DEPARTAMENTAL, TELEFONO 2]
Index: []

Cantidad de establecimientos sin PLAN: 0 siendo el 0.0 % de toda la base de datos


#### Conteo de variantes

In [84]:
df.PLAN.value_counts()

DIARIO(REGULAR)    5365
FIN DE SEMANA      2408
A DISTANCIA          87
SABATINO             49
SEMIPRESENCIAL       26
DOMINICAL            22
MIXTO                 3
INTERCALADO           2
IRREGULAR             2
Name: PLAN, dtype: int64

In [85]:
df.PLAN=df.PLAN.str.title()
df.PLAN

0       Diario(Regular)
1       Diario(Regular)
2       Diario(Regular)
3       Diario(Regular)
4       Diario(Regular)
             ...       
7959    Diario(Regular)
7960      Fin De Semana
7961    Diario(Regular)
7962    Diario(Regular)
7963    Diario(Regular)
Name: PLAN, Length: 7964, dtype: object

## Análisis para variable Departamental

#### Revisión de Nulls

In [86]:
display(df.loc[df.DEPARTAMENTAL.isna(),:].head())
a=len(df.loc[df.DEPARTAMENTAL.isna(),:])
b=len(df)
print('Cantidad de establecimientos sin DEPARTAMENTAL: '+str(a)+' siendo el '+str(round((a/b)*100,2))+' % de toda la base de datos')

Empty DataFrame
Columns: [CODIGO, DISTRITO, DEPARTAMENTO, MUNICIPIO, ESTABLECIMIENTO, DIRECCION, TELEFONO, SUPERVISOR, DIRECTOR, NIVEL, SECTOR, AREA, STATUS, MODALIDAD, JORNADA, PLAN, DEPARTAMENTAL, TELEFONO 2]
Index: []

Cantidad de establecimientos sin DEPARTAMENTAL: 0 siendo el 0.0 % de toda la base de datos


#### Conteo de variantes

In [87]:
df.DEPARTAMENTAL.value_counts()

GUATEMALA NORTE        915
GUATEMALA SUR          665
GUATEMALA OCCIDENTE    661
ESCUINTLA              496
SAN MARCOS             486
HUEHUETENANGO          437
QUETZALTENANGO         430
PETÉN                  357
GUATEMALA ORIENTE      345
SUCHITEPÉQUEZ          330
IZABAL                 303
CHIMALTENANGO          290
ALTA VERAPAZ           280
SACATEPÉQUEZ           276
JUTIAPA                265
RETALHULEU             255
QUICHÉ                 171
CHIQUIMULA             168
SANTA ROSA             140
SOLOLÁ                 138
JALAPA                 116
BAJA VERAPAZ           114
EL PROGRESO            111
ZACAPA                  93
TOTONICAPÁN             81
QUICHÉ NORTE            41
Name: DEPARTAMENTAL, dtype: int64

In [88]:
df.DEPARTAMENTAL=df.DEPARTAMENTAL.str.title()
df.DEPARTAMENTAL

0       Alta Verapaz
1       Alta Verapaz
2       Alta Verapaz
3       Alta Verapaz
4       Alta Verapaz
            ...     
7959          Zacapa
7960          Zacapa
7961          Zacapa
7962          Zacapa
7963          Zacapa
Name: DEPARTAMENTAL, Length: 7964, dtype: object

## Sección Importante (Revisión de establecimientos duplicados)

In [89]:
duplicates=df.loc[df.duplicated(subset=['DISTRITO'
                                        ,'DEPARTAMENTO'
                                        ,'MUNICIPIO'
                                        ,'ESTABLECIMIENTO'
                                        ,'JORNADA'
                                        ,'PLAN'
                                        ,'MODALIDAD'
                                        ,'DIRECCION'
                                        ,'TELEFONO'
                                        ,'SUPERVISOR'
                                        ,'DIRECTOR'
                                        ,'NIVEL'
                                        ,'AREA'
                                        ,'STATUS'
                                        ],keep=False)].copy()
# df.loc[df.duplicated(subset=['CODIGO'],keep=False)]
duplicates=duplicates.sort_values(by=['ESTABLECIMIENTO'])
duplicates

CODIGO DISTRITO    DEPARTAMENTO     MUNICIPIO  \
7139  12-26-0054-46   12-120      San Marcos      Sipacapa   
7138  12-26-0053-46   12-120      San Marcos      Sipacapa   
7178  12-30-0073-46   12-107      San Marcos     La Blanca   
7177  12-30-0072-46   12-107      San Marcos     La Blanca   
6108  14-19-0084-46   14-030          Quiche      Chicaman   
...             ...      ...             ...           ...   
1055  00-01-0644-46   01-627  Ciudad Capital        Zona 1   
1312  00-01-9027-46   01-403  Ciudad Capital        Zona 1   
1232  00-01-8096-46   01-403  Ciudad Capital        Zona 1   
6653  03-12-0038-46   03-007    Sacatepequez  Ciudad Vieja   
6652  03-12-0037-46   03-007    Sacatepequez  Ciudad Vieja   

                                       ESTABLECIMIENTO  \
7139  "Colegio De Estudios Superiores Sipacapa" -Cess-   
7138  "Colegio De Estudios Superiores Sipacapa" -Cess-   
7178       "Colegio Educativo Mixto Luna Azul -Cemla-"   
7177       "Colegio Educativo Mixto Luna Azul -Cemla-"   
6108     "Colegio Mixto Ciencia Y Desarrollo Chicaman"   
...                                                ...   
1055           Liceo Tecnologico Maya De Guatemala Iii   
1312      Liceo Tecnologico Metropolitano De Guatemala   
1232      Liceo Tecnologico Metropolitano De Guatemala   
6653                             Liceo Teresa De Avila   
6652                             Liceo Teresa De Avila   

                             DIRECCION  TELEFONO  \
7139  1Ra. Avenida Y 2Da. Calle Zona 1  33338449   
7138  1Ra. Avenida Y 2Da. Calle Zona 1  33338449   
7178         Parcelamiento Chiquirines  45454204   
7177         Parcelamiento Chiquirines  45454204   
6108             3A. Calle 3-30 Zona 2  58878991   
...                                ...       ...   
1055                     15 Calle 3-29  23083514   
1312                  5A. Avenida 3-55  22327022   
1232                  5A. Avenida 3-55  22327022   
6653           2A. Avenida 2-55 Zona 3  50715990   
6652           2A. Avenida 2-55 Zona 3  50715990   

                            SUPERVISOR  \
7139             Esteban Felix Ramirez   
7138             Esteban Felix Ramirez   
7178           Juan Jose Tobar Tebalan   
7177           Juan Jose Tobar Tebalan   
6108    Jose Alberto Hernandez Gamarro   
...                                ...   
1055    Isaura Amparo Recinos Carrillo   
1312  Carlos Humberto Gonzalez De Leon   
1232  Carlos Humberto Gonzalez De Leon   
6653      Oscar Yovany Zelada Castillo   
6652      Oscar Yovany Zelada Castillo   

                                            DIRECTOR          NIVEL   SECTOR  \
7139                           Baldomero Zapet Rodas  Diversificado  Privado   
7138                           Baldomero Zapet Rodas  Diversificado  Privado   
7178                    German Daniel Merida Joachin  Diversificado  Privado   
7177                    German Daniel Merida Joachin  Diversificado  Privado   
6108                           Wilson Soberaniz Soto  Diversificado  Privado   
...                                              ...            ...      ...   
1055  Margarita De Jesus Barrios Castillo De Godinez  Diversificado  Privado   
1312                              Fredy Amilcar Cruz  Diversificado  Privado   
1232                              Fredy Amilcar Cruz  Diversificado  Privado   
6653          Dora Adilia Teresa Lee Leon De Rosales  Diversificado  Privado   
6652          Dora Adilia Teresa Lee Leon De Rosales  Diversificado  Privado   

        AREA                 STATUS   MODALIDAD JORNADA           PLAN  \
7139  Urbana                Abierta  Monolingue   Doble  Fin De Semana   
7138  Urbana                Abierta  Monolingue   Doble  Fin De Semana   
7178   Rural                Abierta  Monolingue   Doble  Fin De Semana   
7177   Rural                Abierta  Monolingue   Doble  Fin De Semana   
6108  Urbana                Abierta  Monolingue   Doble  Fin De Semana   
...      ...                    ... 

In [90]:
df.to_excel('Proyecto1_DS.xlsx',index=False)